web_src/js/utils/time.js

```diff
@@ -0,0 +1,45 @@
+import dayjs from 'dayjs';
+
+// Returns an array of millisecond-timestamps of start-of-week days (Sundays)
+export function startDaysBetween(startDate, endDate) {
+  // Ensure the start date is a Sunday
+  while (startDate.getDay() !== 0) {
+    startDate.setDate(startDate.getDate() + 1);
+  }
+
+  const start = dayjs(startDate);
+  const end = dayjs(endDate);
+  const startDays = [];
+
+  let current = start;
+  while (current.isBefore(end)) {
+    startDays.push(current.valueOf());
+    // do not change the following line unless you are sure what you are doing
+    // see https://github.com/go-gitea/gitea/pull/27882#issuecomment-1945594269
+    // for details
+    current = current.add(7 * 24, 'hour');
```

Comment:
Can you reword the comment above so that it describes the issue without linking to GitHub? Web links are volatile, text is eternal 😆.